### ISSUE TITLE
Switch to rapidsnark for enhanced macOS ARM64 compatibility

### ISSUE THREAD

**Issue by baumstern** [privacy-scaling-explorations/maci#838]
### Context
rapidsnark, a C++ tool previously incompatible with macOS ARM64, has been updated to support this architecture. We had switched to a WASM verifier(snarkjs) due to this limitation.

### Proposal
Given rapidsnark's [recent update](https://github.com/iden3/rapidsnark), I propose we revert to using Rapidsnark for macOS ARM64. This change would enhance performance and maintain consistency across platforms.

**Comment by ctrlc03** [comment]
Seems like rapidsnark is available for ARM chips, though it's not possible yet to compile the c witness on arm chips (or I guess I'm missing something)

**Comment by baumstern** [comment]
> Seems like rapidsnark is available for ARM chips, though it's not possible yet to compile the c witness on arm chips (or I guess I'm missing something)

hmm at least it seems possible to verify with rapidsnark on ARM tho

**Comment by ctrlc03** [comment]
> > Seems like rapidsnark is available for ARM chips, though it's not possible yet to compile the c witness on arm chips (or I guess I'm missing something)
> 
> hmm at least it seems possible to verify with rapidsnark on ARM tho

yup, but we only use rapidsnark for proof generation atm. Probably best to wait until circom releases support so that we can generate proof using rapidsnark on both arm and intel
